src/main/java/com/exadel/ipromise/exception/UserIsInvalidAuthException.java

```diff
@@ -0,0 +1,7 @@
+package com.exadel.ipromise.exception;
+
+public class UserIsInvalidAuthException extends RuntimeException {
```

Comment:
Instead of NoSuchUserExistsException, UserIsInvalidAuthException and UserAlreadyExistsException you can use one f.e **UserAuthException** and use different messeges depends on situation.